include/proxsuite/proxqp/dense/utils.hpp

```diff
@@ -227,20 +227,26 @@ global_primal_residual_infeasibility(VectorViewMut<T> ATdy,
   // u^T [dz]_+ - l^T[-dz]_+ <= -eps_p_inf ||unscaled(dz)||
   //
   // the variables in entry are changed in place
-  ruiz.unscale_dual_residual_in_place(ATdy);
-  ruiz.unscale_dual_residual_in_place(CTdz);
-  T eq_inf = dy.to_eigen().dot(qpwork.b_scaled);
-  T in_inf = positive_part(dz.to_eigen()).dot(qpwork.u_scaled) -
-             positive_part(-dz.to_eigen()).dot(qpwork.l_scaled);
-  ruiz.unscale_dual_in_place_eq(dy);
-  ruiz.unscale_dual_in_place_in(dz);
-
-  T bound_y = qpsettings.eps_primal_inf * infty_norm(dy.to_eigen());
-  T bound_z = qpsettings.eps_primal_inf * infty_norm(dz.to_eigen());
-
-  bool res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
-             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z &&
-             infty_norm(dy.to_eigen()) != 0 && infty_norm(dz.to_eigen()) != 0;
+
+  bool res = infty_norm(dy.to_eigen()) != 0 && infty_norm(dz.to_eigen()) != 0;
+  if (!res) {
+    return res;
+  } else {
+
+    ruiz.unscale_dual_residual_in_place(ATdy);
+    ruiz.unscale_dual_residual_in_place(CTdz);
+    T eq_inf = dy.to_eigen().dot(qpwork.b_scaled);
+    T in_inf = positive_part(dz.to_eigen()).dot(qpwork.u_scaled) -
+               positive_part(-dz.to_eigen()).dot(qpwork.l_scaled);
+    ruiz.unscale_dual_in_place_eq(dy);
+    ruiz.unscale_dual_in_place_in(dz);
+
+    T bound_y = qpsettings.eps_primal_inf * infty_norm(dy.to_eigen());
+    T bound_z = qpsettings.eps_primal_inf * infty_norm(dz.to_eigen());
+
+    res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
+          infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z;
+  }
```

Comment:
You should remove the else here and put the content in the main scope of the function.